src/contexts/ProcedureProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import { createContext, ReactNode, useCallback, useContext, useMemo, useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+
+export const categories = ["Economy", "Housing", "Employment", "Justice"] as const;
+
+export interface Component {
+  id: string;
+  name: string;
+  properties: object;
+  required?: string[];
+}
+
+export interface Procedure {
+  name: string;
+  category: typeof categories[number];
+  description: string;
+  components: Component[];
+}
+
+interface Context {
+  procedure: Procedure | null;
+  createProcedure: (procedure: Procedure) => void;
+  addComponent: (component: Omit<Component, "id">) => void;
+  deleteComponentById: (id: string) => void;
+}
+
+const ProcedureContext = createContext<Context>({} as Context);
+export const useProcedures = () => useContext(ProcedureContext);
+
+export const ProcedureProvider = ({ children }: { children: ReactNode }) => {
+  const [procedure, setProcedure] = useState<Procedure | null>(null);
+
+  const createProcedure = useCallback((procedure: Procedure) => {
+    setProcedure({ ...procedure, components: [] });
+  }, []);
```

Comment:
I'd call this "resetProcedure", because there seems to be only 1 procedure, and this function erases all its components?